tests/auditd/test_auditd.py

```diff
@@ -0,0 +1,272 @@
+                   tacacs_creds['tacacs_rw_user'],
+                   tacacs_creds['tacacs_rw_user_passwd'],
+                   "sudo docker ps")
+    cmd = """show logging | grep 'audisp-syslog' | grep 'docker_commands' | grep 'AUID="test_rwuser"' """
+    result = duthost.shell(cmd)['stdout_lines']
+    assert len(result) > 0, "Auditd docker_commands rule does not contain the expected logs"
+
+
+def test_auditd_config_changes(localhost, duthosts, enum_rand_one_per_hwsku_hostname, tacacs_creds, check_tacacs): # noqa F811
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+    dutip = duthost.mgmt_ip
+    container_name = "auditd"
+    verify_container_running(duthost, container_name)
+
+    watch_files = {
+        "group_changes": ["/etc/group"],
+        "hosts_changes": ["/etc/hosts"],
+        "passwd_changes": ["/etc/passwd"],
+        "shadow_changes": ["/etc/shadow"],
+        "sudoers_changes": ["/etc/sudoers"],
+        "time_changes": ["/etc/localtime"],
+        "auth_logs": ["/var/log/auth.log",
+                      "/var/log.tmpfs/auth.log"],
+        "cron_changes": ["/etc/crontab",
+                         "/etc/cron.d",
+                         "/etc/cron.daily",
+                         "/etc/cron.hourly",
+                         "/etc/cron.weekly",
+                         "/etc/cron.monthly"],
+        "dns_change": ["/etc/resolv.conf"],
+        "docker_config": ["/etc/docker/daemon.json"],
+        "docker_daemon": ["/usr/bin/dockerd"],
+        "docker_service": ["/lib/systemd/system/docker.service"],
+        "docker_socket": ["/lib/systemd/system/docker.socket"],
+        "modules_changes": ["/sbin/insmod",
+                            "/sbin/rmmod",
+                            "/sbin/modprobe"],
+        "log_changes": ["/var/log/testfile",
+                        "/var/log.tmpfs/testfile"],
+        "bin_changes": ["/bin/testfile"],
+        "sbin_changes": ["/sbin/testfile"],
+        "usr_bin_changes": ["/usr/bin/testfile"],
+        "usr_sbin_changes": ["/usr/sbin/testfile"],
+        "docker_storage": ["/var/lib/docker/testfile"]
+    }
+
+    for rule, files in watch_files.items():
+        for file in files:
+            ssh_remote_run(localhost,
+                           dutip,
+                           tacacs_creds['tacacs_rw_user'],
+                           tacacs_creds['tacacs_rw_user_passwd'],
+                           f"sudo touch {file}")
+            cmd = f"""show logging | grep 'audisp-syslog' | grep '{rule}' | grep 'AUID="test_rwuser"' """
+            result = duthost.shell(cmd)['stdout_lines']
+            assert len(result) > 0, f"Auditd {rule} rule does not contain the expected logs"
+
+
+def test_auditd_host_failure(localhost, duthosts, enum_rand_one_per_hwsku_hostname):
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+    container_name = "auditd_watchdog"
+    verify_container_running(duthost, container_name)
+
+    # Simulate auditd service failure
```

Comment:
>Simulate [](http://example.com/codeflow?start=6&length=8)

better to kill the service?

Reply:
auditd will auto restart when it got killed

Reply:
Understood. Seem you  remove stop auditd in latest code, will you still simulate audit failure?

Reply:
Yes I moved it to conftest.py file. The test cases are unchanged, they still have auditd failure simulation